cloud/team/team.go

```diff
@@ -0,0 +1,708 @@
+	ErrWrongEnforceInput        = errors.New("the input to the `--enforce-cicd` flag")
+	ErrNoShortName              = errors.New("cannot retrieve organization short name from context")
+	ErrNoTeamsFoundInOrg        = errors.New("no teams found in your organization")
+	ErrNoTeamsFoundInWorkspace  = errors.New("no teams found in your workspace")
+	ErrNoTeamMembersFoundInTeam = errors.New("no team members found in team")
+	ErrNoUsersFoundInOrg        = errors.New("no users found in your organization")
+	teamPagnationLimit          = 100
+)
+
+func CreateTeam(name, description string, out io.Writer, client astrocore.CoreClient) error {
+	if name == "" {
+		return ErrInvalidName
+	}
+	ctx, err := context.GetCurrentContext()
+	if err != nil {
+		return err
+	}
+	if ctx.OrganizationShortName == "" {
+		return user.ErrNoShortName
+	}
+	teamCreateRequest := astrocore.CreateTeamJSONRequestBody{
+		Description: &description,
+		Name:        name,
+	}
+	resp, err := client.CreateTeamWithResponse(httpContext.Background(), ctx.OrganizationShortName, teamCreateRequest)
+	if err != nil {
+		return err
+	}
+	err = astrocore.NormalizeAPIError(resp.HTTPResponse, resp.Body)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(out, "Astro Team %s was successfully created\n", name)
+	return nil
+}
+
+func UpdateWorkspaceTeamRole(id, role, workspace string, out io.Writer, client astrocore.CoreClient) error {
+	err := user.IsWorkspaceRoleValid(role)
+	if err != nil {
+		return err
+	}
+	ctx, err := context.GetCurrentContext()
+	if err != nil {
+		return err
+	}
+	if ctx.OrganizationShortName == "" {
+		return user.ErrNoShortName
+	}
+	if workspace == "" {
+		workspace = ctx.Workspace
+	}
+
+	teams, err := GetWorkspaceTeams(client, workspace, teamPagnationLimit)
+	if err != nil {
+		return err
+	}
+	if len(teams) == 0 {
+		return ErrNoTeamsFoundInWorkspace
+	}
+	var team astrocore.Team
+	if id == "" {
+		team, err = selectTeam(teams)
+		if err != nil {
+			return err
+		}
+	} else {
+		for i := range teams {
+			if teams[i].Id == id {
+				team = teams[i]
+			}
+		}
+
+		if team.Id == "" {
+			return ErrTeamNotFound
+		}
+	}
+	teamID := team.Id
+
+	teamMutateRequest := astrocore.MutateWorkspaceTeamRoleRequest{Role: role}
+	resp, err := client.MutateWorkspaceTeamRoleWithResponse(httpContext.Background(), ctx.OrganizationShortName, workspace, teamID, teamMutateRequest)
+	if err != nil {
+		return err
+	}
+	err = astrocore.NormalizeAPIError(resp.HTTPResponse, resp.Body)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(out, "The workspace team %s role was successfully updated to %s\n", teamID, role)
+	return nil
+}
+
+func UpdateTeam(id, name, description string, out io.Writer, client astrocore.CoreClient) error {
+	ctx, err := context.GetCurrentContext()
+	if err != nil {
+		return err
+	}
+	if ctx.OrganizationShortName == "" {
+		return user.ErrNoShortName
+	}
+	teams, err := GetOrgTeams(client)
```

Comment:
similarly in here to get the org team directly with team ID.

Reply:
done